runtime/sema/type.go

```diff
@@ -5591,6 +5709,19 @@ func checkSubTypeWithoutEquality(subType Type, superType Type) bool {
 			return typedSubType.ExplicitInterfaceConformanceSet().
 				Contains(typedSuperType)
 
+		// An interface type is a supertype of a restricted type if the restricted set contains
+		// that explicit interface type. Once interfaces can conform to interfaces, this should instead
+		// check that at least one value in the restriction set is a subtype of the interface supertype
+
+		// This particular case comes up when checking attachment access; enabling the following expression to typechecking:
+		// resource interface I { /* ... */ }
+		// attachment A for I { /* ... */ }
+		//
+		// let i : &{I} = ... // some operation constructing `i`
```

Comment:
I think the the rule is actually for `{I}`, not `&{I}`:

```suggestion
		// let i : {I} = ... // some operation constructing `i`
```